### ISSUE TITLE
[Ingest] Change document field of _simulate with verbose=true to diff view

### ISSUE THREAD

**Issue by talevy** [elastic/elasticsearch#14698]
This is in regards to the `_simulate` endpoint.

It would be nice to have a view of what was modified in a document between processors in a pipeline instead of seeing a whole new copy of the full document.

Currently, the output for a pipeline looks like this

```
{
      "error": false,
      "processor_results": [
        {
          "processor_id": "processor[mutate]-0",
          "error": false,
          "modified": true,
          "doc": {
            "_index": "index",
            "_type": "type",
            "_source": {
              "bar": "hello",
              "foo": 5
            },
            "_id": "id2"
          }
        },
        {
          "processor_id": "processor[mutate]-1",
          "error": false,
          "modified": true,
          "doc": {
            "_index": "index",
            "_type": "type",
            "_source": {
              "bar": "HELLO",
              "foo": 5
            },
            "_id": "id2"
          }
        }
      ]
    }
```

From this, you can see that the `doc` field shows all of the document's fields. It would be great to only show what has been modified from after the previous processor's mutations.

Something like this would be nice:

```
{
      "error": false,
      "processor_results": [
        {
          "processor_id": "processor[mutate]-0",
          "error": false,
          "modified": true,
          "doc_diff": {
            "updated": {
              "_source.foo": 5
            },
            "added": {},
            "removed": {}
          }
        },
        ...
      ]
    }
```

depends on: https://github.com/elastic/elasticsearch/pull/14572

**Comment by talevy** [comment]
implementation reference from Guava: [source](https://github.com/google/guava/blob/8ff490732db7a71adcf561171b5444ddc61c8990/guava/src/com/google/common/collect/Maps.java#L428) [docs](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/MapDifference.html)

**Comment by clintongormley** [comment]
@talevy is this something you're still planning on investigating?

**Comment by talevy** [comment]
@clintongormley I will go ahead and close it. This responsibility/feature has been pushed down to clients to implement. For example, Kibana computes diffs using the current complete response documents.
